### ISSUE TITLE
convert_to_async IsAwaitableType prefix-matches TaskCompletionSource/TaskFactory/… → CS1061

### ISSUE THREAD

**Issue by JoshuaRamirez** [JoshuaRamirez/RoslynMcpServer#1163]
## Summary

`convert_to_async` classifies awaitable return types with a **display-string prefix** check (`StartsWith("System.Threading.Tasks.Task")`). That matches real `Task` / `Task<T>`, but also every other type whose display name starts the same way — none of which are awaitable. `FindAwaitableCalls` then selects those invocations, eligibility treats the method as convertible, and the rewriter emits `await …` → **CS1061** (no `GetAwaiter`) while claiming **SUCCESS**.

| Return type (display) | Product `IsAwaitableType` | Emitted | Error |
| --- | --- | --- | --- |
| `System.Threading.Tasks.TaskCompletionSource<int>` | **True** | `await Make();` | **CS1061** |
| `System.Threading.Tasks.TaskFactory` | **True** | `await Make();` | **CS1061** |
| `System.Threading.Tasks.TaskScheduler` | **True** | `await Make();` | **CS1061** |
| `System.Threading.Tasks.TaskCanceledException` | **True** | `await Make();` | **CS1061** |
| `System.Threading.Tasks.TaskStatus` | **True** | (same if invoked) | **CS1061** |
| `System.Threading.Tasks.Task` / `Task<int>` | True | `await Inner();` | clean (control) |

Same prefix also false-positives `TaskCreationOptions`, `TaskContinuationOptions`, etc. Real `Task`/`Task<T>`/`ValueTask` controls remain fine.

## Repro

Master `d3abad3d5c215daf0728afd209d4f11ef906d8e4` (Release **v0.6.2**, packages Roslyn **5.0.0**).

```csharp
using System.Threading.Tasks;
class C {
  TaskCompletionSource<int> MakeTcs() => new();
  TaskFactory MakeFac() => Task.Factory;
  TaskScheduler MakeSched() => TaskScheduler.Default;
  TaskCanceledException MakeEx() => new TaskCanceledException();
  Task Inner() => Task.CompletedTask;
  Task<int> InnerT() => Task.FromResult(1);
  void UseTcs() { MakeTcs(); }
  void UseFac() { MakeFac(); }
  void UseSched() { MakeSched(); }
  void UseEx() { MakeEx(); }
  void UseTask() { Inner(); }
  void UseTaskT() { var x = InnerT(); }
}
```

1. `convert_to_async` on `UseTcs` → **SUCCESS** with body `await MakeTcs();` → compile **CS1061**.
2. Same on `UseFac` / `UseSched` / `UseEx` → **CS1061**.
3. Control: `UseTask` → `await Inner();` compiles.
4. Control: `UseTaskT` → `var x = await InnerT();` compiles.

Live probe (Roslyn **5.0.0**): product `IsAwaitableType` + annotate + `AnnotatedCallRewriter` VisitInvocationExpression await wrap + `ToAsyncReturnType` copied from master `d3abad3…` — TCS / TaskFactory / TaskScheduler / TaskCanceledException → **PRODUCT_WRONG=yes** (CS1061); Task / Task\<int\> controls → **WRONG=no**. Display dump: all five non-awaitable `System.Threading.Tasks.Task*` types report `IsAwaitable=True`.

## Expected

- Only truly awaitable types (`Task`, `Task<T>`, `ValueTask`, `ValueTask<T>`, or a general GetAwaiter-based / INotifyCompletion check) should be selected by `FindAwaitableCalls`.
- Methods whose only “Task*” calls return `TaskCompletionSource` / `TaskFactory` / `TaskScheduler` / `TaskCanceledException` / similar should either:
  - skip those invocations (and throw `NoAsyncCalls` if none remain), **or**
  - refuse conversion with a clear error — never SUCCESS + `await` of a non-awaitable.
- Real `Task` / `Task<T>` / `ValueTask` awaits stay as today.
- Do not leave SUCCESS + CS1061.

## Actual

| Shape | Product body | Rewrite | Expected |
| --- | --- | --- | --- |
| `void M() { MakeTcs(); }` | `await MakeTcs();` SUCCESS | **CS1061** | skip / refuse |
| `void M() { var x = MakeTcs(); }` | `var x = await MakeTcs();` | **CS1061** | skip / refuse |
| `void M() { MakeFac(); }` | `await MakeFac();` | **CS1061** | skip / refuse |
| `void M() { MakeSched(); }` | `await MakeSched();` | **CS1061** | skip / refuse |
| `void M() { MakeEx(); }` | `await MakeEx();` | **CS1061** | skip / refuse |
| `void M() { Inner(); }` (`Task`) | `await Inner();` | clean | unchanged intent |
| `void M() { var x = InnerT(); }` (`Task<int>`) | `var x = await InnerT();` | clean | unchanged intent |

## Root cause

`IsAwaitableType` (~L1160–1168) uses:

```csharp
typeName.StartsWith("System.Threading.Tasks.Task") ||
typeName.StartsWith("System.Threading.Tasks.ValueTask") || …
```

`TaskCompletionSource<T>`, `TaskFactory`, `TaskScheduler`, `TaskCanceledException`, `TaskStatus`, `TaskCreationOptions`, `TaskContinuationOptions`, etc. all share the `System.Threading.Tasks.Task` prefix, so they are treated as awaitable.

`FindAwaitableCalls` (~L1137–1157) adds those invocations; eligibility (~L170–175 / ~L524–525) accepts `awaitableCalls.Count > 0`; `AnnotatedCallRewriter.VisitInvocationExpression` (~L1279–1304) always wraps annotated calls in `await`. The result is SUCCESS with illegal C#.

This is the **false-positive** side of string-prefix awaitable detection. `#1153` Separate follow-ups item 2 noted the complementary **false-negative** (custom awaitables not named Task/ValueTask) and suggested a general `IsAwaitable` check — that same fix would close this class.

## Acceptance criteria

- [ ] Methods returning / invoking `TaskCompletionSource` / `TaskFactory` / `TaskScheduler` / `TaskCanceledException` (and other non-awaitable `Task*` types) are not awaited
- [ ] If those are the only candidates, conversion fails with `NoAsyncCalls` (or an explicit clear error) — not SUCCESS + CS1061
- [ ] `Task` / `Task<T>` / `ValueTask` / `ValueTask<T>` still convert and await as today
- [ ] Prefer symbol-based awaitable detection (original definition Task/ValueTask **or** GetAwaiter pattern) over `StartsWith("…Task")`
- [ ] allFiles uses the same rule
- [ ] Regression tests for TCS / TaskFactory / TaskScheduler / TaskCanceledException + Task / Task\<T\> controls
- [ ] Does not claim to fix `#1160` var/discard CS0815, `#1156` Wait/Result/GetResult, `#1153` Task\<Task\> double-wrap, DocumentNotEditable `#552`, reserved `#571` unless deliberately batched

## Out of scope

- `var` / discard await of non-generic Task/ValueTask → CS0815/CS8209 — `#1160`
- Sync-block `.Wait` / `.Result` / `GetAwaiter().GetResult()` — `#1156`
- Already-Task return double-wrap (`Task` → `Task<Task>`) — `#1153`
- Custom awaitables **false negatives** (types with GetAwaiter but not named Task/ValueTask) — `#1153` FU2; file separately if live WRONG after a prefix-only fix that does not add GetAwaiter detection
- Leftover extract work — label **bug** only

## Separate follow-ups (noticed, NOT filed)

1. **Typed `Task t = await NonGenericTask()` / argument-position `Sink(await NonGenericTask())`** — CS0029/CS1503 siblings of `#1160` var/discard (live likely WRONG; file separately if still open after `#1160`).
2. **Custom awaitables false negatives** (GetAwaiter present, name ≠ Task/ValueTask) — `#1153` FU2; only file if live WRONG after this prefix fix if the fix does not add general awaitable detection.
3. **`.ConfigureAwait(false).GetAwaiter().GetResult()`** — only file if live WRONG after `#1156` (`#1160` FU1 / `#1156` FU2).
4. **DocumentNotEditable / reserved / allFiles+line** for this tool remain `#552` / `#571` / `#489`.

## Named from

Live hunt on master `d3abad3…` after `#1160` Separate follow-ups were exhausted (FU1 deferred on `#1156`; FU2 already `#552`/`#571`/`#489`). Related to `#1153` Separate follow-ups item 2 (general `IsAwaitable` vs string prefix) — this filing is the **false-positive** Task\*-prefix class, live-verified **WRONG=yes**.

## Evidence footer

- Master: `d3abad3d5c215daf0728afd209d4f11ef906d8e4` (Extract MethodInterfaceHelpers.ImplementsInterface `#1161`/`#1162`; Release **v0.6.2**)
- Packages: Roslyn **5.0.0** / product **0.6.2**
- Product path: `IsAwaitableType` ~L1160–1168; `FindAwaitableCalls` ~L1137–1157; eligibility ~L170–175 / ~L524–525; `AnnotatedCallRewriter.VisitInvocationExpression` ~L1279–1304
- Live probe: Roslyn **5.0.0** — product prefix check + annotate + await wrap — TCS / TaskFactory / TaskScheduler / TaskCanceledException → CS1061 **WRONG=yes**; Task / Task\<int\> **WRONG=no**; display strings confirm prefix false positives
- Open leftover at filing: **0** — label **bug** only; not leftover (not exclusive-end class)
- Closest (orthogonal): `#1160` var/discard void await CS0815/CS8209; `#1156` Wait/Result/GetResult CS4008/CS1061 on real Tasks; `#1153` Task\<Task\> double-wrap / return await CS0029 — none is convert_to_async **SUCCESS + CS1061** from awaiting non-awaitable `System.Threading.Tasks.Task*` types via prefix match
- Leave Dependabot untouched
